crates/router/src/core/unified_connector_service.rs

```diff
@@ -399,3 +435,283 @@ pub fn handle_unified_connector_service_response_for_payment_repeat(
+    .attach_printable("Failed to build UCS webhook transform request")?;
+
+    // Build connector auth metadata
+    let connector_auth_metadata = merchant_connector_account
+        .map(|mca| {
+            #[cfg(feature = "v1")]
+            let mca_type = MerchantConnectorAccountType::DbVal(Box::new(mca.clone()));
+            #[cfg(feature = "v2")]
+            let mca_type = MerchantConnectorAccountTypeDetails::MerchantConnectorAccount(Box::new(
+                mca.clone(),
+            ));
+
+            build_unified_connector_service_auth_metadata(mca_type, merchant_context)
+        })
+        .transpose()
+        .change_context(ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to build UCS auth metadata")?
+        .ok_or_else(|| {
+            error_stack::report!(ApiErrorResponse::InternalServerError).attach_printable(
+                "Missing merchant connector account for UCS webhook transformation",
+            )
+        })?;
+
+    // Build gRPC headers
+    let grpc_headers = external_services::grpc_client::GrpcHeaders {
+        tenant_id: state.tenant.tenant_id.get_string_repr().to_string(),
+        request_id: Some(crate::utils::generate_id(consts::ID_LENGTH, "webhook_req")),
+    };
+
+    // Make UCS call
+    if let Some(ucs_client) = &state.grpc_client.unified_connector_service_client {
+        match ucs_client
+            .transform_incoming_webhook(transform_request, connector_auth_metadata, grpc_headers)
+            .await
+        {
+            Ok(response) => {
+                let transform_response = response.into_inner();
+                let transform_data =
+                    transformers::transform_ucs_webhook_response(transform_response)
+                        .change_context(ApiErrorResponse::InternalServerError)
+                        .attach_printable("Failed to handle UCS webhook transform response")?;
+
+                // UCS handles everything internally - event type, source verification, decoding
+                Ok((
+                    transform_data.event_type,
+                    transform_data.source_verified,
+                    transform_data,
+                ))
+            }
+            Err(err) => {
+                // When UCS is configured, we don't fall back to direct connector processing
+                // since the goal is to remove direct connector code in the future
+                router_env::logger::error!(
+                    error = ?err,
+                    "UCS webhook transformation failed for UCS-enabled merchant/connector"
+                );
+                Err(ApiErrorResponse::WebhookProcessingFailure)
+                    .attach_printable(format!("UCS webhook processing failed: {err}"))
+            }
+        }
+    } else {
+        // UCS client not available but UCS is configured
+        // We don't fall back to direct connector processing when UCS is configured
+        router_env::logger::error!(
```

Comment:
Attach printable already server the logging purpose, so this log will be redundant one